File: tests/js/sentry-test/fixtures/incident.js
import {IncidentRule} from './incidentRule';

export function Incident(params) {
  return {
    id: '321',
    identifier: '123',
    organizationId: '3',
    dateClosed: '2019-04-19T19:44:05.963Z',
    dateStarted: '2019-04-05T19:44:05.963Z',
    dateDetected: '2019-04-05T19:44:05.963Z',
    dateCreated: '2019-04-05T19:44:05.963Z',
    title: 'Too many Chrome errors',
    status: 0,
    projects: [],
    isSubscribed: true,
    alertRule: IncidentRule(),
    ...params,
  };
}

File: src/sentry/incidents/endpoints/serializers/incident.py
from datetime import datetime
from typing import TypedDict

from sentry.api.serializers.models.incidentactivity import IncidentActivitySerializerResponse
from sentry.incidents.endpoints.serializers.alert_rule import (
    AlertRuleSerializerResponse,
)


class IncidentSerializerResponse(TypedDict):
    id: str
    identifier: str
    organizationId: str
    projects: list[str]
    alertRule: AlertRuleSerializerResponse
    activities: list[IncidentActivitySerializerResponse] | None
    status: int
    statusMethod: int
    type: int
    title: str
    dateStarted: datetime
    dateDetected: datetime
    dateCreated: datetime
    dateClosed: datetime | None


class DetailedIncidentSerializerResponse(IncidentSerializerResponse):
    discoverQuery: str
